tests/foreman/cli/test_product.py

```diff
@@ -335,6 +337,34 @@ def test_positive_product_sync_state():
         3. Successfully sync another repository under the same product.
         4. Run `hammer product info --product-id <id>` again.
 
+
     :expectedresults: hammer should show 'Sync Incomplete' in both cases.
     """
-    pass
+    product = make_product({'organization-id': module_org.id})
+    repo_a1 = make_repository(
+        {
+            'organization-id': module_org.id,
+            'product-id': product['id'],
+            'name': gen_string('alpha'),
+            'url': f'{gen_url(scheme="https")}:{gen_integer(min_value=10, max_value=9999)}',
+        }
+    )
+    with pytest.raises(CLIReturnCodeError):
```

Comment:
Which of the 4 statements do you expect to raise a `CLIReturnCodeError`?

I suspect its not the last statement, the assertion, which suggests some of these lines aren't executing during the test.

The `with` catches and handles the exception type passed to it, but doesn't change the fact that the lines inside of its context after the exception is raised do not execute. When the expected exception occurs, the context is exited.

The line which raises the exception should always be the last line of the `pytest.raises` context.

https://docs.pytest.org/en/6.2.x/reference.html#pytest-raises

Reply:
On L(353) when the sync occurs as it's a dummy URL the metadata cannot be fetched and surely will improvise the pytest.raises with that only.